### ISSUE TITLE
Updating symbology according to field value 

### ISSUE THREAD

**Issue by NiceStitch** [3liz/lizmap-web-client#4588]
### Is your feature request related to a problem? Please describe
QGIS expressions don't seem to be working to change the symbology of an object according to a given value. I've used a concatenation to choose a raster image as a raster marker, it works in QGIS but not in Lizmap, even after changing the path according to the docs.

It goes as follow in QGIS on Windows = @project_home || '\\\media\\\panneau\\\\' || "code_panneau" || '.png' 

I've put the images in a media folder on server and I've changed it for Lizmap to = '../media/panneau/ || "code_panneau" || '.png' 

It doesn't work no matter how I try

### Describe the solution you'd like

For it to wok just like on Windows, when symbology changes according to the value of "code_panneau".

### Describe alternatives you've considered
For the moment I've changed the symbology from Single symbol to Categorized, but it's a live PostGIS layer, updated every date with Qfield so it's not ideal to do it by hand.

**Comment by Gustry** [comment]
Can you add some screenshots ? (in the symbology tab and the result in the legend panel)

> even after changing the path according to the docs.

Can you give the link ?

**Comment by NiceStitch** [comment]
Yeah, of course. So in QGIS on Windows it goes like this 
![Capture d'écran 2024-07-09 122344](https://github.com/3liz/lizmap-web-client/assets/146181611/965fab8b-5348-44ca-acd5-6882cca8d3aa)
with the result being : 
![Capture d'écran 2024-07-09 122958](https://github.com/3liz/lizmap-web-client/assets/146181611/96c9c57c-0381-4385-8a2b-3169824d63b9)

I've changed it for Lizmap to : 
![Capture d'écran 2024-07-09 122213](https://github.com/3liz/lizmap-web-client/assets/146181611/7a6212fe-0ca7-4803-91dd-dc6b401aeb03)
but the result goes as this : 
![Capture d'écran 2024-07-09 122611](https://github.com/3liz/lizmap-web-client/assets/146181611/495dbe5d-4299-4dc7-aee7-26dcf132ff04)

I've tried to change it to ../media etc. but it just doesn't change. It's either a question mark or nothing.

**Comment by Gustry** [comment]
It works for me.
You can debug by adding a label for instance with `file_exists`.

Can you give the link to the docs you are mentioning ?
Can you do the right click step https://docs.lizmap.com/current/en/publish/lizmap_plugin/information.html#request-for-support

**Comment by NiceStitch** [comment]
I've mostly used this part : https://docs.lizmap.com/3.7/en/publish/configuration/media.html

My Lizmap version is :
 **Versions :**

* Lizmap Web Client : 3.7.8
* Lizmap plugin : 4.3.15
* QGIS Desktop : 3.22.5
* QGIS Server : 3.22.4
* Py-QGIS-Server : not used
* QGIS Server plugin atlasprint : 3.4.1
* QGIS Server plugin cadastre : 1.19.1
* QGIS Server plugin lizmap_server : 2.9.2
* QGIS Server plugin wfsOutputExtension : not found

<details>
<summary>List of safeguards :</summary>
<br/>

* Mode : normal
* Allow parent folder : no
* Prevent other drive : yes
* Prevent PG service : yes
* Prevent PG Auth DB : no
* Force PG user&pass : yes
* Prevent ECW : yes
</details>

**Comment by Gustry** [comment]
> QGIS Desktop : 3.22.5
QGIS Server : 3.22.4

Both versions are very outdated.
3.22 is not maintained for a long time. 3.22.5 was released February 2022...

Try to upgrade first : https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule

**Comment by NiceStitch** [comment]
I was suspecting that it might be the issue, sadly I don't have control over the server to update it, I'll contact the IT departement. Thank you for your help and for a confirmation that it's supposed to work correctly

**Comment by Gustry** [comment]
Strange, you are running a up to date LWC version, but QGIS server is very old :/

I tried on 3.34.8 both server and desktop.

BTW, do not forget to update your QGIS desktop plugin as well.

**Comment by NiceStitch** [comment]
I think I've closed the subject too fast. After upgrading everything the bug persists and the symbology doesn't update automatically. I've tried too follow your advice to debug with file_exists but the result is just confusing 
![Capture d'écran 2024-07-10 164638](https://github.com/3liz/lizmap-web-client/assets/146181611/a6fd2aa0-6573-4274-92cb-caf8b3e0fc51)
Both use the same path '../media/panneau/AB4.png' 
I've tried every combination to get it working, and even tried to move files directly to media folder, but everything comes back as false. I've checked with is_directory too to see if it at least recognizes the subdirectory /panneau but it's not the case.

Do you have any advice ? Can you share your configuration for when you tested it yesterday? 

Thanks in advance for your help

**Comment by Gustry** [comment]
You can debug with your label and the expression used in your [first screenshot](https://github.com/3liz/lizmap-web-client/issues/4588#issuecomment-2217285698).

Also use a single `/` ? Is your server on windows ? And this makes it works also on Windows I think.

**Comment by NiceStitch** [comment]
I've tried that already, the path is created correctly. The server runs on Ubuntu.

I just don't understand why the concatenation works as intended. It can display a singular image if I specify the path manually. But the moment I use a function it all crumbles. is_directory returns "../media/" as true but "../media/ panneau/" as false and any file check with is_file or file_exists returns false no matter the placement.

I've used double \ in QGIS desktop running on Windows as it would consider it as skip otherwise.
